src/path-alias.ts

```diff
@@ -0,0 +1,82 @@
+import * as vscode from "vscode";
+import { WORKSPACE_FOLDER_VARIABLE } from "./constants";
+import { PathAlias } from "./options";
+import { memoize } from "lodash";
+import * as JSON5 from "json5";
+import * as fse from "fs-extra";
+
+function valueContainsWorkspaceFolder(value: string): boolean {
+  return value.indexOf(WORKSPACE_FOLDER_VARIABLE) >= 0;
+}
+
+function filterWorkspaceFolderAlias(pathAlias: PathAlias): PathAlias {
+  const newAlias: PathAlias = {};
+  for (const key in pathAlias) {
+    if (!valueContainsWorkspaceFolder(pathAlias[key])) {
+      newAlias[key] = pathAlias[key];
+    }
+  }
+  return newAlias;
+}
+
+function getBaseUrlFromTsConfig(tsconfig: {
+  compilerOptions: { baseUrl: string };
+}): string {
+  const baseUrl = tsconfig?.compilerOptions?.baseUrl;
+  if (baseUrl) {
+    return baseUrl.startsWith("./") ? baseUrl.replace("./", "") : baseUrl;
+  }
+
+  return "";
+}
+
+function replaceWorkspaceFolderWithRootPath(
+  pathAlias: PathAlias,
+  rootPath: string
+): PathAlias {
+  const newAlias: PathAlias = {};
+  for (const key in pathAlias) {
+    newAlias[key] = pathAlias[key].replace(WORKSPACE_FOLDER_VARIABLE, rootPath);
+  }
+
+  return newAlias;
+}
+
+const getTsAlias = memoize(async function (
+  workfolder: vscode.WorkspaceFolder
+): Promise<PathAlias> {
+  const include = new vscode.RelativePattern(workfolder, "[tj]sconfig.json");
+  const exclude = new vscode.RelativePattern(workfolder, "**/node_modules/**");
+  const files = await vscode.workspace.findFiles(include, exclude);
+
+  const mapping: PathAlias = {};
+  for (let i = 0; i < files.length; i++) {
+    try {
+      const fileContent = await fse.readFile(files[i].fsPath, "utf8");
+      const configFile = JSON5.parse(fileContent);
+      const baseUrl = getBaseUrlFromTsConfig(configFile);
+      if (baseUrl) {
+        mapping[baseUrl] = WORKSPACE_FOLDER_VARIABLE + "/" + baseUrl;
```

Comment:
Not support paths property ?
> Yep, `paths property` is also **TODO** for PathIntellisense
https://github.com/ChristianKohler/PathIntellisense/blob/1103971e5eb8b147f8a76302e8861534fc79162b/src/configuration/tsconfig.service.ts#L49

Reply:
tsconfig set paths alias config
```js
{
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "@clinyong": ["./src/clinyong/*"]  // <- support this
    }
  }
}
```
css modules use paths alias
```jsx
import * as s from '@clinyong/button.module.scss';
```

Reply:
em... still working in progress...

Reply:
As we support the paths property, the `baseUrl` seems not necessary, so I remove it.